engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -883,6 +894,18 @@ case class JesBackend(actorSystem: ActorSystem)
     lazy val jesStderrGcsPath = callGcsPath.resolve(jesLogStderrFilename(key)).toString
 
     val callContext = new CallContext(callGcsPath.toString, jesStdoutGcsPath, jesStderrGcsPath)
-    new JesCallEngineFunctions(workflowDescriptor.ioManager, callContext)
+    new JesCallEngineFunctions(workflowDescriptor.fileSystems, callContext)
+  }
+
+  override def fileSystems(options: WorkflowOptions, workflowRootPath: String): List[FileSystem] = {
+    // TODO Should GCS operations have their own execution context ?
+    implicit val executionContext = scala.concurrent.ExecutionContext.global
```

Comment:
I'm going to punt on this for a future ticket... but for discussion:

Yes, they should have either their own `ExecutionContext`, or it should be passed in. If I'm reading correctly, `GoogleCloudStorageWriteChannel` is using an `ExecutionContext` for async upload, only during `GcsFileSystemProvider.newOutputStream(Path)`. But afaik, this still synchronously returns an `OutputStream` that is then used synchronously by the caller. Meanwhile, none of the `FileSystem` operations externally behave asynchronously. Thus if GCS behind the scenes has it's own asynchonicity going on, it should, IMHO, have it's own, internal, thread pool. Also it cleans it up such that these _O(n)_ callers don't need to keep passing in a pool. If the GCS wants to use the "global" pool internally, then it should.

IF on the other hand, the `FileSystem` operations were returning `Future[_]` asynchronous results, then they should also be taking in an execution context as a (probably implicit) parameter.